internal/log/glg/glg.go

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package glg
+
+import (
+	logger "github.com/kpango/glg"
+	"github.com/vdaas/vald/internal/log/retry"
+)
+
+type GlgLogger struct {
+	lv    level
+	rout  retry.Out
+	routf retry.Outf
+	log   *logger.Glg
+}
+
+// New returns a new GlgLogger instance.
+func New(g *logger.Glg, opts ...Option) *GlgLogger {
+	gl := (&GlgLogger{
+		log: g,
+	}).apply(append(defaultOpts, opts...)...)
+
+	gl.setLevelMode(gl.lv)
+	return gl
+}
+
+func Default() *GlgLogger {
+	gl := (&GlgLogger{
+		log: logger.Get(),
+	}).apply(defaultOpts...)
```

Comment:
I can't agree with this coding pattern eg. apply function

setting FOP should be located in Constructor func Default is not constructor, please use func New and initialize once using sync.Once